From a18564c819a0fb8cbc26a7a59e8d6ff0e38ddb57 Mon Sep 17 00:00:00 2001
From: Daniel Dragomir <Daniel.Dragomir@windriver.com>
Date: Mon, 7 Oct 2019 14:01:45 +0000
Subject: [PATCH] build.sh: use -DNDEBUG also in debug builds

Without it, there is a significant performance regression
when running 'bjam install'.

The idea for this fix came from upstream, commit 270a33863e
("boost: update to 1.71.0"), but it was adapted since 1.70
sources (at least build.sh) are quite different than 1.71.

Signed-off-by: Daniel Dragomir <Daniel.Dragomir@windriver.com>
---
 tools/build/src/engine/build.jam | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/build/src/engine/build.jam b/tools/build/src/engine/build.jam
index de567ddb5..f61a2d011 100644
--- a/tools/build/src/engine/build.jam
+++ b/tools/build/src/engine/build.jam
@@ -217,7 +217,7 @@ toolset darwin cc :  "-o " : -D
 toolset gcc gcc : "-o " : -D
     : -pedantic -fno-strict-aliasing
     [ opt --release : [ opt --symbols : -g : -s ] -O3 ]
-    [ opt --debug : -g -O0 -fno-inline ]
+    [ opt --debug : -g -O0 -fno-inline -DNDEBUG ]
     [ opt --profile : -O3 -g -pg ]
     -I$(--python-include) -I$(--extra-include) -Wno-long-long
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
-- 
2.17.1

